client.go

```diff
@@ -1408,6 +1408,25 @@ func linkProperty(key string, value interface{}) LinkOption {
 	}
 }
 
+// LinkSourceName sets the name of the source queue
+func LinkSourceName(name string) LinkOption {
+	return func(l *link) error {
+		l.name = name
```

Comment:
I'd like to have some way of preventing duplicate names. It's not possible here, but it can be done during handle allocation.

You can check if the name already exists in the `linksByName` map and set an error on the link if it does (lines 523-525 are an example of setting an error). This should happen before `handles.next()` is called.

https://github.com/vcabbage/amqp/blob/61c744178624aae38c323c668e1a25a5995f6dd3/client.go#L519-L530

Reply:
It would be nice to add a test for the duplicate name behavior, if you feel up to it. Don't feel obligated to do so though.

Reply:
Fixed

Reply:
The current integration tests all rely on Microsoft Service Bus and Event Hubs. They're a bit of a pain to run unless you already have an Azure account that you don't mind giving the tests wide ranging access to. Some integration tests for other brokers that can be run in docker would be nice.

There are a couple issues with the test and I forgot that the entry in `linksByName` gets deleted after the attach returns. The fix isn't necessarily difficult, but since you're unable to run the tests it might be easier for me to resolve. Do you mind if I push some fixes to this PR? I might not get a chance to do it immediately since I'm at work and have a meeting to go to.

Reply:
Thank you for helping me! I have updated this PR with your changes.